repository/src/main/resources/alfresco/public-services-security-context.xml

```diff
@@ -32,12 +32,12 @@
             <ref bean="dictionaryService" />
         </property>
     </bean>
-    
+
     <!-- ======================= -->
     <!-- Support for permissions -->
     <!-- ========================-->
 
-    <bean id='permissionService' class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
+    <bean id="permissionService" class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean" depends-on="transactionManager, permissionServiceImpl">
```

Comment:
How come we had to specify this `depends-on` clause when the 2 beans are already included as refs to set in the bean's properties?

Reply:
After tens of hours of debugging circular dependencies and nulls inside `transactionManager`, trying to understand what exactly Spring has change in bean creation algorithm, we came up with this solution to force the correct order by adding `depends-on`. 
It looked like Spring was trying to create `permissionService` before the beans that are injected into it.